src/onboarding/steps.test.ts

```diff
@@ -0,0 +1,254 @@
+
+describe('onboarding steps', () => {
+  const newUserFlowWithEverythingEnabled = {
+    onboardingProps: {
+      ...mockOnboardingProps,
+      skipVerification: false,
+      supportedBiometryType: BIOMETRY_TYPE.FACE_ID,
+      recoveringFromStoreWipe: false,
+    },
+    screens: [
+      Screens.NameAndPicture,
+      Screens.PincodeSet,
+      Screens.EnableBiometry,
+      Screens.VerificationStartScreen,
+    ],
+    name: 'newUserFlowWithEverythingEnabled',
+  }
+
+  const newUserFlowWithEverythingDisabled = {
+    onboardingProps: {
+      ...mockOnboardingProps,
+      skipVerification: true,
+      supportedBiometryType: null,
+      recoveringFromStoreWipe: false,
+    },
+    screens: [Screens.NameAndPicture, Screens.PincodeSet],
+    name: 'newUserFlowWithEverythingDisabled',
+  }
+
+  const importWalletFlowEverythingEnabled = {
+    onboardingProps: {
+      ...mockOnboardingProps,
+      recoveringFromStoreWipe: false,
+      choseToRestoreAccount: true,
+      skipVerification: false,
+      supportedBiometryType: BIOMETRY_TYPE.FACE_ID,
+    },
+    screens: [
+      Screens.NameAndPicture,
+      Screens.PincodeSet,
+      Screens.EnableBiometry,
+      Screens.ImportWallet,
+      Screens.VerificationStartScreen,
+    ],
+    name: 'importWalletFlowEverythingEnabled',
+  }
+
+  it.each([
+    newUserFlowWithEverythingEnabled,
+    newUserFlowWithEverythingDisabled,
+    importWalletFlowEverythingEnabled,
+  ])(
+    'goToNextOnboardingScreen and getOnboardingStepValues work as expected',
+    ({ onboardingProps, screens, name }) => {
+      const expectedTotalSteps = screens.length
+      screens.forEach((screen, index) => {
+        const { totalSteps, step } = getOnboardingStepValues(screen, onboardingProps)
+        // Checking that the step number is correct
+        try {
+          expect(step).toEqual(index + 1)
+          expect(totalSteps).toEqual(expectedTotalSteps)
+        } catch (error) {
+          throw new Error(
+            `Expected step ${
+              index + 1
+            } and totalStep ${expectedTotalSteps} but got ${step} and ${totalSteps} for screen ${screen} in test ${name}`
+          )
+        }
+
+        goToNextOnboardingScreen({ firstScreenInCurrentStep: screen, onboardingProps })
+
+        // If we are on the last screen, we should navigate home
+        // If we are not on the last screen, we should navigate to the next screen via navigate or navigateClearingStack
+        if (index === screens.length - 1) {
+          // eslint-disable-next-line jest/no-conditional-expect
+          expect(navigateHome).toHaveBeenCalled()
+        } else {
+          try {
+            // eslint-disable-next-line jest/no-conditional-expect
+            expect(navigate).toHaveBeenCalledWith(screens[index + 1])
+          } catch {
+            // eslint-disable-next-line jest/no-conditional-expect
+            expect(navigateClearingStack).toHaveBeenCalledWith(screens[index + 1])
+          }
```

Comment:
```suggestion
                expect(index === screens.length - 1 ? navigateHome : ((navigate as jest.MockedFunction<any>).mock.calls.length ? navigate : navigateClearingStack)).toHaveBeenCalled()
```
this is pretty ugly but I think it works

Reply:
and it lets us not disable linting

Reply:
oh just realized it means we can't use toHaveBeenCalledWith. Ugh. your original is probly better.